instrumented/nightly-tests/src/androidTest/kotlin/com/datadog/android/nightly/trace/SpanConfigE2ETests.kt

```diff
@@ -0,0 +1,259 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.nightly.trace
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.LargeTest
+import androidx.test.platform.app.InstrumentationRegistry
+import com.datadog.android.Datadog
+import com.datadog.android.core.configuration.Configuration
+import com.datadog.android.event.SpanEventMapper
+import com.datadog.android.nightly.SPECIAL_STRING_TAG_NAME
+import com.datadog.android.nightly.rules.NightlyTestRule
+import com.datadog.android.nightly.utils.initializeSdk
+import com.datadog.android.nightly.utils.measure
+import com.datadog.android.nightly.utils.measureSdkInitialize
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.tracing.AndroidTracer
+import com.datadog.android.tracing.model.SpanEvent
+import fr.xgouchet.elmyr.junit4.ForgeRule
+import io.opentracing.util.GlobalTracer
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+@LargeTest
+class SpanConfigE2ETests {
+
+    @get:Rule
+    val forge = ForgeRule()
+
+    @get:Rule
+    val nightlyTestRule = NightlyTestRule()
+
+    /**
+     * apiMethodSignature: Credentials#constructor(String, String, String, String?, String? = null)
+     */
+    @Test
+    fun trace_config_feature_enabled() {
+        val testMethodName = "trace_config_feature_enabled"
+        measureSdkInitialize {
+            initializeSdk(
+                InstrumentationRegistry.getInstrumentation().targetContext,
+                config = Configuration.Builder(
+                    logsEnabled = true,
+                    tracesEnabled = true,
+                    rumEnabled = true,
+                    crashReportsEnabled = true
+                ).build()
+            )
+        }
+        GlobalTracer.get()
+            .buildSpan(testMethodName)
+            .start()
+            .finish()
```

Comment:
What's the purpose of this zero-duration span? Shouldn't we send `testMethodName` span from `measureSdkInitialize {}`?

Reply:
`measureSdkInitialize` will have its own span (called `sdk_initialize`). Since we are testing trace feature we want to make sure that span is sent when feature is enabled, and we want to have dedicated span in order not to mix assertions in the monitors with spans coming from other other places.

Also we don't want to put anything else in the `measureSdkInitialize` other than SDK initialization logic itself, because then we would measure something else.

Reply:
Very clear, thanks 🙂!